eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/UnmodifiableBiMap.java

```diff
@@ -1133,25 +1133,48 @@ public void clear()
     @Override
     public V getIfAbsentPut(K key, Function0<? extends V> function)
     {
-        throw new UnsupportedOperationException("Cannot call getIfAbsentPut() on " + this.getClass().getSimpleName());
+        V result = this.get(key);
+        if (this.isAbsent(result, key))
+        {
+            throw new UnsupportedOperationException("Cannot mutate " + this.getClass().getSimpleName());
+        }
+        return result;
```

Comment:
Implementations like these are debatable, but are done consistently elsewhere like in UnmodifiableMutableMap. The trait-based testing approach is applying identical assertions to all the unmodifiable map wrappers and exposing differences like these.